src/libraries/System.Private.Uri/src/System/Uri.cs

```diff
@@ -2708,312 +2710,246 @@ private string GetUnescapedParts(UriComponents uriParts, UriFormat formatAs)
         private string ReCreateParts(UriComponents parts, ushort nonCanonical, UriFormat formatAs)
         {
             EnsureHostString(false);
-            string stemp = (parts & UriComponents.Host) == 0 ? string.Empty : _info.Host!;
-            // we reserve more space than required because a canonical Ipv6 Host
-            // may take more characters than in original m_String
-            // Also +3 is for :// and +1 is for absent first slash
-            // Also we may escape every character, hence multiplying by 12
-            // UTF-8 can use up to 4 bytes per char * 3 chars per byte (%A4) = 12 encoded chars
-            int count = (_info.Offset.End - _info.Offset.User) * (formatAs == UriFormat.UriEscaped ? 12 : 1);
-            char[] chars = new char[stemp.Length + count + _syntax.SchemeName.Length + 3 + 1];
-            count = 0;
+            string str = _string;
+
+            ValueStringBuilder dest = str.Length <= StackallocThreshold
+                ? new ValueStringBuilder(stackalloc char[StackallocThreshold])
```

Comment:
It looks like ReCreateParts might call other methods that in turn stackalloc as well?  How much stack space might we end up trying to use?

Reply:
Should be max `2 * StackallocThreshold`, so currently 2048 bytes at any given time.